kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/ConsolidatedKafkaStreamsApp.java

```diff
@@ -0,0 +1,139 @@
+package org.hypertrace.core.kafkastreams.framework;
+
+import com.typesafe.config.Config;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.kstream.KStream;
+import org.hypertrace.core.serviceframework.config.ConfigClient;
+import org.hypertrace.core.serviceframework.config.ConfigUtils;
+
+/**
+ * Base class for consolidating multiple independent KafkaStreamApps
+ */
+public abstract class ConsolidatedKafkaStreamsApp extends KafkaStreamsApp {
+
+  static final String SUB_TOPOLOGY_NAMES_CONFIG_KEY = "sub.topology.names";
+  static final String ENV_CLUSTER_NAME_KEY = "cluster.name";
+  static final String ENV_POD_NAME_KEY = "pod.name";
+  static final String ENV_CONTAINER_NAME_KEY = "container.name";
+
+  private Map<String, Pair<String, KafkaStreamsApp>> jobNameToSubTopology = new HashMap<>();
```

Comment:
For readability, instead of Pair `Pair<String, KafkaStreamsApp>>`, shall we have custom static class here say:

```
static class SubTopologyKStramApp {
  String subTopologyName;
  KafkaStreamsApp instance;
}
```
So, we can have:
```
private Map<String, SubTopologyKStramApp> jobNameToSubTopology = new HashMap<>();
```

What do you think?

Reply:
I just saw that's how we used in hypertrace-ingester as well.

Reply:
have made the change